velox/functions/sparksql/tests/MapTest.cpp

```diff
@@ -105,5 +105,41 @@ TEST_F(MapTest, ErrorCases) {
       {inputVector1, inputVector2, inputVector11, inputVector22},
       true);
 }
+
+TEST_F(MapTest, ComplexTypes) {
```

Comment:
Is it supported to create map from array? Maybe we can add test for that if it's enabled.

Reply:
I think there is another function for array: https://spark.apache.org/docs/latest/api/sql/#map_from_arrays

Reply:
But let me confirm if array is valid input for this function, thanks.

Reply:
Added ut, thanks.
![image](https://github.com/facebookincubator/velox/assets/10524738/f40ef223-32f4-4f35-9877-d2b98b64da5a)